tools/compute_accuracy.py

```diff
@@ -0,0 +1,238 @@
+    img_set -- the file containing a list of images
+    resize_mode -- the mode used to resize images
+
+    Keyword arguments:
+    oversampling -- boolean, True if oversampling should be enabled
+    """
+
+    img_matrix = pd.read_csv(img_set, engine='python', header=None,sep=r" (?!(.*) )")
+
+    mean_blob = caffe.proto.caffe_pb2.BlobProto()
+    mean_data = open(mean_file, 'rb').read()
+    mean_blob.ParseFromString(mean_data)
+    mean = np.array(caffe.io.blobproto_to_array(mean_blob))[0].mean(1).mean(1)
+
+    # Loading the classifier
+    caffe.set_mode_gpu()
+    if grayscale:
+        net = Classifier(deploy_file, snapshot,
+                       mean=mean,
+                       raw_scale=255)
+    else:
+        net = Classifier(deploy_file, snapshot,
+                       mean=mean,
+                       channel_swap=(2,1,0),
+                       raw_scale=255)
+
+    size = len(img_matrix)
+    num_classes = np.array(img_matrix[2])[-1]+1
+    prediction = np.zeros(size, dtype=int)
+    labels = np.zeros(size, dtype=int)
+    probas = np.zeros([size, num_classes], dtype=float)
+ 
+    for i in range(0, size): 
+        cur_image = img_matrix[0][i]
+        if grayscale:
+            input_image = utils.image.load_image(cur_image)            
+            input_image = utils.image.resize_image(input_image, width, height, resize_mode=resize_mode, channels=1)
+            input_image = input_image.astype(np.uint8)
+            input_image = input_image.reshape(input_image.shape[0], input_image.shape[1], 1)
+        else:
+            input_image = utils.image.load_image(cur_image)
+            input_image = utils.image.resize_image(input_image, width, height, resize_mode=resize_mode)
+            input_image = skimage.img_as_float(input_image).astype(np.float32)
+
+        labels[i] = img_matrix[2][i]
+        probas[i] = net.predict([input_image], oversample=False)
+        prediction[i] = probas[i].argmax()
+        if size > 500 and i % (size/500) == 0:
+            logger.debug("Progress: %0.2f" % (i/float(size)))
+
+    snapshot_file, snapshot_extension = os.path.splitext(snapshot)
+
+    # Dumping the result
+    joblib.dump(probas, snapshot_file + "-accuracy-proba.pkl")
+    joblib.dump(labels, snapshot_file + "-accuracy-labels.pkl")
+    logger.debug("Done")
+    return True
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Accuracy-Computing tool - DIGITS')
+    parser.add_argument('snapshot',
+            help='The Caffe model snapshot.'
+            )
+
+    parser.add_argument('deploy_file',
+            help='The deploy_file.'
+            )
+
+    parser.add_argument('labels',
+            help='The file containing the labels.'
+            )
+
+    parser.add_argument('mean_file',
+            help='The dataset mean file.'
+            )
+    parser.add_argument('img_set',
+            help='The file representing the set of files to evaluate'
+            )
```

Comment:
Why aren't you passing the prebuilt database instead of the textfile? Those images are already loaded and resized. Then all you would need to know is the crop size.